addons/point_of_sale/models/product_combo.py

```diff
@@ -0,0 +1,15 @@
+from odoo import api, fields, models, _
+
+
+class ProductCombo(models.Model):
+    _name = "product.combo"
+    _description = "Product combo choices"
+
+    name = fields.Char(string="Name", required=True)
+    combo_product_ids = fields.One2many("product.combo.item", "combo_id", string="Products in Combo")
+    no_of_products = fields.Integer("No of Products", compute="_compute_no_of_products")
+
+    @api.depends("combo_product_ids")
+    def _compute_no_of_products(self):
+        for rec in self:
+            rec.no_of_products = len(rec.combo_product_ids)
```

Comment:
explore using read_group to this value. Imagine you're in a list view of product.combo with many items, and it's slow because of "no_of_products". I know it's an optimization. But it's nicer to start with something that is fast already. Besides, it's good to learn read_group.

Reply:
Sure, I will, but this `len` will look at the number of products that are in the `combo` and this means that it should only ever have a couple of them at any one time. That being said, i will do the optimisation